### ISSUE TITLE
Stackoverflow exception when writing data set

### ISSUE THREAD

**Issue by lellid** [Apollo3zehn/PureHDF#86]
Hi,
I have the following code:

```
var len = 3;
var dataArr = new double[len];
var dataset = new H5Dataset(dataArr, fileDims: [(ulong)len]);
var file = new H5File()
{
  ["raw"] = dataset,
};
file.Write(fileName);
```
(The reason why I used an `H5Dataset` instead of an array is that later on I want to add attributes to the dataset, and modify the dimensions)
The code shows a StackoverflowException when it reaches `file.Write`, in the method `H5NativeWriter.EncodeDataset`. Is this a bug, or did I use something in the wrong way?

Best regards, Dirk

**Comment by Apollo3zehn** [comment]
Which version of PureHDF are you running and on which OS? With the current version the code

```cs
var len = 3;
var dataArr = new double[len];
var dataset = new H5Dataset(dataArr, fileDims: [(ulong)len]);
var file = new H5File()
{
    ["raw"] = dataset,
};

file.Write("/home/vincent/Downloads/out.h5");
```

runs well for me on Ubuntu 24.04. 

Thanks :-)

**Comment by lellid** [comment]
It succeeds unter .Net8.05, but fails with a StackoverflowException under NetFramework 4.8.9232.0, using Windows 10.0.19045, with the version of PureHDF of 1.0.0-beta.16. An excerpt of the stack is:

```
System.Memory.dll!System.SpanHelpers.CopyTo<byte>(ref byte dst, int dstLength, ref byte src, int srcLength)
System.Memory.dll!System.Span<byte>.TryCopyTo(System.Span<byte> destination)	
System.Memory.dll!System.Span<byte>.ToArray()	
PureHDF.dll!PureHDF.StreamExtensions.Write(System.IO.Stream stream, System.Span<byte> buffer)	
PureHDF.dll!PureHDF.StreamExtensions.Write(System.IO.Stream stream, System.Span<byte> buffer)	
[The 1 frame(s) above this were repeated 12612 times]	
PureHDF.dll!PureHDF.StreamExtensions.Write(System.IO.Stream stream, System.Span<byte> buffer)	
PureHDF.dll!PureHDF.VFD.H5StreamDriver.Write(System.Span<byte> data)	
PureHDF.dll!PureHDF.VOL.Native.ObjectHeader2.Encode(PureHDF.VOL.Native.NativeWriteContext context)	
PureHDF.dll!PureHDF.H5NativeWriter.InternalEncodeDataset<double[], double>(PureHDF.H5Dataset dataset, double[] data, bool isScalar)
``` 

Hope that helps.

Best regards, Dirk

**Comment by Apollo3zehn** [comment]
Thanks for that hint, I will try to reproduce it!

**Comment by Apollo3zehn** [comment]
The bug should have been solved in version v1.0.0-beta.17 :-)

**Comment by lellid** [comment]
Yes, it works now! Thanks a lot!
